internal/app/backend/backend_service.go

```diff
@@ -51,21 +66,118 @@ func newBackend(cfg config.View) (*backendService, error) {
 // streams these results back to the caller.
 func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Backend_FetchMatchesServer) error {
 	ctx := stream.Context()
+	var c <-chan *pb.Match
+	switch interface{}(req.Config.Type).(type) {
+	// MatchFunction Hosted as a GRPC service
+	case *pb.FunctionConfig_Grpc:
+		client, err := s.getGrpClient(interface{}(req.Config.Type).(*pb.FunctionConfig_Grpc))
+		if err != nil {
+			logger.WithFields(logrus.Fields{
+				"error":    err.Error(),
+				"function": req.Config,
+			}).Error("failed to connect to match function")
+			return status.Error(codes.InvalidArgument, "failed to connect to match function")
+		}
+
+		// Get the channel over which the generated match results will be sent.
+		c = s.matchesFromGrpcMMF(ctx, req.Profile, client)
+
+	// MatchFunction Hosted as a REST service
+	case *pb.FunctionConfig_Rest:
+		return status.Error(codes.Unimplemented, "not implemented")
+	}
 
 	for {
 		select {
 		case <-ctx.Done():
-			return nil
+			return ctx.Err()
 
-		default:
-			err := stream.Send(&pb.FetchMatchesResponse{})
+		case match, ok := <-c:
+			if !ok {
+				// Channel closed indicating that we are done processing matches.
+				return nil
+			}
+
+			// Send the match result over output channel.
+			m, ok := proto.Clone(match).(*pb.Match)
+			if !ok {
+				logger.Error("failed to clone match proto")
+				return status.Error(codes.Internal, "failed processing match result")
+			}
+
+			err := stream.Send(&pb.FetchMatchesResponse{Match: m})
 			if err != nil {
 				return err
 			}
 		}
 	}
 }
 
+func (s *backendService) getGrpClient(config *pb.FunctionConfig_Grpc) (pb.MatchFunctionClient, error) {
+	addr := config.Host + ":" + strconv.FormatInt(int64(config.Port), 10)
+	_, ok := s.mmfClients[addr]
```

Comment:
Because this can run on multiple fibers you'll need to protect this with a mutex or use sync.Map